.github/workflows/deploy-prod.yml

```diff
@@ -18,12 +18,59 @@ jobs:
     environment: production
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'maven'
 
       - name: Set Repository Name
         id: repo_name
         run: |
           echo "name=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_OUTPUT
 
+      - name: SonarQube Scan and Quality Gate
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_PROJECT_KEY: ${{ secrets.SONAR_PROJECT_KEY }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        run: |
+          ./mvnw sonar:sonar \
+            -Dsonar.projectKey=${{ secrets.SONAR_PROJECT_KEY }} \
+            -Dsonar.host.url=${{ secrets.SONAR_HOST_URL }} \
+            -Dsonar.token=${{ secrets.SONAR_TOKEN }}
+          
+          # Quality Gate Check
+          SONAR_CE_TASK_URL=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?componentKey=${{ secrets.SONAR_PROJECT_KEY }}")
+          
+          TASK_ID=$(echo $SONAR_CE_TASK_URL | jq -r '.task.id')
+          
+          # Wait for task to complete and get analysis ID
+          ANALYSIS_ID=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
+          
+          # Check quality gate status
+          QUALITY_GATE_STATUS=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/qualitygates/project_status?analysisId=$ANALYSIS_ID" | \
+          jq -r '.projectStatus.status')
+          
+          echo "Quality Gate Status: $QUALITY_GATE_STATUS"
+          
+          if [ "$QUALITY_GATE_STATUS" != "OK" ]; then
+            echo "❌ Quality Gate Failed"
+            exit 1
+          fi
+
```

Comment:
This works in theory, but when i searched online i found out that there exists an official github action from sonar themself which does the quality gate checking, see:
https://github.com/SonarSource/sonarqube-quality-gate-action?tab=readme-ov-file#usage
This has more error handling and better status reporting.